src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -13898,8 +13891,28 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
 
         case INS_lea:
             // uops.info
-            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
-            result.insLatency    = PERFSCORE_LATENCY_1C;
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X; // one or two components
+                result.insLatency    = PERFSCORE_LATENCY_1C;
+
+                regNumber indexReg = id->idAddr()->iiaAddrMode.amIndxReg;
+                if (indexReg != REG_NA)
+                {
+                    regNumber baseReg = id->idAddr()->iiaAddrMode.amBaseReg;
+                    if (baseReg != REG_NA)
+                    {
+                        ssize_t dsp = emitGetInsAmdAny(id);
+                        if (dsp != 0)
+                        {
+                            result.insThroughput = PERFSCORE_THROUGHPUT_1C; // three components
```

Comment:
I don't believe that a three component  LEA has a latency of 3, in Agner
I have used the instruction timings from from https://uops.info/table.html 
 LEA_BID (R64)  https://uops.info/html-instr/LEA_BID_R64.html

Reply:
In any case the issue rate of LEA  (0.5 or 1.0) is always worse than the add instruction (0.25), so we should avoid using LEA instructions.

Reply:
I found this info on LEA on the llvm.org  site:

https://reviews.llvm.org/D32277

According to Intel's Optimization Reference Manual for SNB+:
" For LEA instructions with three source operands and some specific situations, instruction latency has increased to 3 cycles, and must
  dispatch via port 1:
- LEA that has all three source operands: base, index, and offset
- LEA that uses base and index registers where the base is EBP, RBP,or R13
- LEA that uses RIP relative addressing mode
- LEA that uses 16-bit addressing mode "

I can change the latency to 3 cycles, for the three operand case, and can open an issue to migrate away from using LEA instruction.

Reply:
That information is much more illuminating. I couldn't find a latency on either of the links you indicate, but in the instruction_tables.pdf it has a latency of 3 for "3 components in address rip relative address" (page 255). I find that table a bit confusing because it's not clear whether the lack of a latency in a row implies that it's the same as above, or simply that it's not provided (like in the case of the rip relative address).